### ISSUE TITLE
find and subsitute not match case

### ISSUE THREAD

**Issue by SomaLily** [replit/codemirror-vim#195]
![image](https://github.com/user-attachments/assets/0d96f493-2788-4440-8a77-bcd72124821e)

This can be confusing when one wants to replace keys.

**Comment by firai** [comment]
Not entirely sure I'm looking at the right parts of the code, but it seems like `ignoreCase` and `smartCase` may have hardcoded values at various places, but the options are in the functions? So maybe new options to change the parameters would be needed?
https://github.com/replit/codemirror-vim/blob/59c44ef2909506836bb1fd2f81c738e35b29a4aa/src/vim.js#L1671
https://github.com/replit/codemirror-vim/blob/59c44ef2909506836bb1fd2f81c738e35b29a4aa/src/vim.js#L1702-L1703
https://github.com/replit/codemirror-vim/blob/59c44ef2909506836bb1fd2f81c738e35b29a4aa/src/vim.js#L1741
https://github.com/replit/codemirror-vim/blob/59c44ef2909506836bb1fd2f81c738e35b29a4aa/src/vim.js#L1790

**Comment by firai** [comment]
It seems like the defaults in codemirror-vim are different than the vim defaults right now. vim defaults for both [`ignorecase`](https://vimhelp.org/options.txt.html#%27ignorecase%27) and [`smartcase`](https://vimhelp.org/options.txt.html#%27smartcase%27) are both off, while codemirror-vim default seems to behave as if `smartcase` on, which in turn implies that `ignorecase` is also on.

**Comment by nightwing** [comment]
Yes, i think turning smarcase on by default was an intentional choice of the author of vim mode for codemirror5 https://github.com/codemirror/codemirror5/issues/2302#issuecomment-37511905

By default we have javascript regex mode turned on, and it is more natural in this mode to have a default that is common in other modern editors, the people who want exactly vim behavior who already have to change the setting for pcre can change `ic` setting as well .

Perhaps we should change the text on the right side to 3 buttons `pcre  ic  scs`?
